src/chronify/sqlalchemy/functions.py

```diff
@@ -1,37 +1,45 @@
+from typing import Literal, TypeAlias
 import pandas as pd
 import polars as pl
 from numpy.dtypes import ObjectDType
 from pandas import DatetimeTZDtype
 from sqlalchemy import Connection, Selectable
 
-from chronify.time_configs import DatetimeRange
+from chronify.time_configs import DatetimeRange, TimeBaseModel
+
+# Copied from Polars, which doesn't export the type.
+DbWriteMode: TypeAlias = Literal["replace", "append", "fail"]
 
 
 def read_database(
-    query: Selectable | str, conn: Connection, config: DatetimeRange
+    query: Selectable | str, conn: Connection, config: TimeBaseModel
```

Comment:
This is what I had before, but we only accept DatetimeRange though...

Reply:
Not anymore. We have to be able to read a table that was registered in representative time.